packages/transitive-overlay/src/util.ts

```diff
@@ -323,10 +348,10 @@ export function itineraryToTransitive(
       // Flex routes sometimes have the same from and to IDs, but
       // these stops still need to be rendered separately!
       if (isFlex(leg)) {
-        if (typeof leg.to.stop === "object") {
-          leg.to.stop.id = `${leg.to.stop.id}_flexed_to`;
+        if (typeof modifiedToStop.stop === "object") {
+          modifiedToStop.stop.id = `${modifiedToStop.stop.id}_flexed_to`;
         } else {
-          leg.to.stopId = `${leg.to.stopId}_flexed_to`;
+          modifiedToStop.stopId = `${modifiedToStop.stopId}_flexed_to`;
         }
       }
```

Comment:
This block is covered in new function `cloneToStop` and should now be removed from here.